File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    cards1 = cards1[::-1]
    cards2 = cards2[::-1]
    
    for v in goal:
        if cards1 and cards1[-1] == v:
            cards1.pop()
        elif cards2 and cards2[-1] == v:
            cards2.pop()
        else:
            return 'No'
        
    return 'Yes'

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int p1 = 0, p2 = 0;
        
        for (String word : goal) {
            if (p1 < cards1.length && cards1[p1].equals(word)) p1++;
            else if (p2 < cards2.length && cards2[p2].equals(word)) p2++;
            else return "No";
        }
        return "Yes";
    }
}
